opm/simulators/flow/FlowProblemComp.hpp

```diff
@@ -290,13 +361,29 @@ class FlowProblemComp : public FlowProblem<TypeTag>
             fs.setViscosity(FluidSystem::gasPhaseIdx, FluidSystem::viscosity(fs, paramCache, FluidSystem::gasPhaseIdx));
         }
 
+        // determine the component fractions
+        Dune::FieldVector<Scalar, numComponents> z(0.0);
+        Scalar sumMoles = 0.0;
+        for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
+            for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+                Scalar tmp = Opm::getValue(fs.molarity(phaseIdx, compIdx) * fs.saturation(phaseIdx));
```

Comment:
Here's another instance of `fs.saturation()`  which could be hoisted outside the `compIdx` loop.